### PR TITLE
Update copyright and file header guidance for .NET Core

### PR DESCRIPTION
**Pull request by richlander**
- Update copyright in LICENSE.txt.
- Add file header guidance.
- Clean up contributing.md a bit.

Answers to a few likely questions ...

Q: Is Microsoft going to move to this new file header?
A: Yes. Files will be switched over. No timeline has been set for that and not part of this change. New files from Microsoft will use the new file header.

Q: Changing the copyright is usually considered a big deal! How's that work?
A: The project was assigned to the .NET Foundation by Microsoft (~ November 2014). The copyright statement has been corrected to match that.

Q: You are removing signs of "Microsoft" from the project. Does that mean Microsoft is leaving the project?
A: Nope. Not even close.

Q: I would like a way to be recognized for my contributions beyond just commits. I was planning on using copyright statements for that, but that's no longer an option with the new file header guidance.
A: We are planning on using release notes to call out contributors for a given release.

Q: There are some open PRs with new files in them that don't align with this guidance.
A: Those PRs will need to be updated before merging.

Q: Will this change apply to more than just CoreCLR?
A: It will apply to all of .NET Core.

Q: Is it raining in Seattle right now?
A: Yes.

### CODE REVIEW

File: Documentation/project-docs/contributing.md [hunk 1]
```diff
@@ -99,13 +70,62 @@ DOs and DON'Ts
 * **DON'T** submit PRs that alter licensing related files or headers. If you believe there's a problem with them, file an issue and we'll be happy to discuss it.
 * **DON'T** add API additions without filing an issue and discussing with us first. See [API Review Process](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md).
 
+Contribution "Bar"
+==================
+
+Project maintainers will merge changes that align with [project priorities](project-priorities.md) and/or improve the product significantly for a broad set of apps. Proposed must also satisfy the published [guidelines for .NET Core](#contribution-guidelines).
+
+Maintainers will not merge changes that have narrowly-defined benefits, due to compatibility risk. The CoreCLR codebase is used by several Microsoft products (e.g. Windows Phone, ASP.NET 5, .NET Framework 4.6) to enable execution of managed code. Changes to the open source codebase can become part of these products, but are first reviewed and tested to ensure they are correct for those products and will not inadvertently break applications. We may revert changes if they are found to be breaking.
+
```
Comment (shahid-pk) [inline Documentation/project-docs/contributing.md]: nit : ASP.NET 5 is now ASP.NET CORE

**Comment by RichiCoder1** [comment]
> Q: Is it raining in Seattle right now?
A: Yes.

Q: Is it really though?

File: Documentation/project-docs/contributing.md [hunk 2]
```diff
@@ -99,13 +70,62 @@ DOs and DON'Ts
 * **DON'T** submit PRs that alter licensing related files or headers. If you believe there's a problem with them, file an issue and we'll be happy to discuss it.
 * **DON'T** add API additions without filing an issue and discussing with us first. See [API Review Process](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md).
 
+Contribution "Bar"
+==================
+
+Project maintainers will merge changes that align with [project priorities](project-priorities.md) and/or improve the product significantly for a broad set of apps. Proposed must also satisfy the published [guidelines for .NET Core](#contribution-guidelines).
```
Comment (akoeplinger) [inline Documentation/project-docs/contributing.md]: ~~Proposed~~ Proposals

**Comment by richlander** [comment]
@RichiCoder1 Yes.

**Pull request merged**